pkg/shardddl/optimism/column.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package optimism
+
+import (
+	"go.etcd.io/etcd/clientv3"
+
+	"github.com/pingcap/dm/dm/common"
+	"github.com/pingcap/dm/pkg/etcdutil"
+)
+
+// GetAllDroppedColumns gets the all dropped columns.
+func GetAllDroppedColumns(cli *clientv3.Client) (map[string]map[string]map[string]map[string]map[string]interface{}, int64, error) {
```

Comment:
why the final value of that big map is `interface{}` not `struct{}`

Reply:
What's the difference? What will we benefit if we use `struct{}`?

Reply:
compared to interface{}, struct{} uses no memory, and don't need a type assertion when assign (don't know if compiler will eliminate this)

and in rest of DM code we use struct{} as map value